python_modules/dagster/dagster/_config/structured_config/__init__.py

```diff
@@ -101,6 +99,14 @@ class Config(MakeConfigCacheable):
     Base class for Dagster configuration models.
     """
 
+    @property
+    def config_dictionary(self) -> Mapping[str, Any]:
+        """
+        Returns a dictionary representation of this config object,
+        ignoring any private fields.
+        """
+        return {k: v for k, v in self.__dict__.items() if not k.startswith("_")}
+
```

Comment:
seems like a dangerous precedent to make this a public without prefix. Recommend a naming convention similar to `_asdict()` on named tuples. How about `_as_config_dict`?